File: examples/PHP/durasub.php
<?php
/*
 *  Durable subscriber
 */

$context = new ZMQContext(1);

//  Connect our subscriber socket
$subscriber = new ZMQSocket($context, ZMQ::SOCKET_SUB);
$subscriber->setSockOpt(ZMQ::SOCKOPT_IDENTITY, "Hello");
$subscriber->setSockOpt(ZMQ::SOCKOPT_SUBSCRIBE, "");
$subscriber->connect("tcp://localhost:5565");


//  Synchronize with publisher
$sync = new ZMQSocket($context, ZMQ::SOCKET_PUSH);
$sync->connect("tcp://localhost:5564");
$sync->send("");

//  Get updates, expect random Ctrl-C death
while(true) {
	$string = $subscriber->recv();
	echo $string, "\n";
	if($string == "END") {
		break;
	}
}

File: examples/Perl/durasub.pl
#!/usr/bin/perl
=pod

Durable subscriber

Author: Alexander D'Archangel (darksuji) <darksuji(at)gmail(dot)com>

=cut

use strict;
use warnings;
use 5.10.0;

use ZeroMQ qw/:all/;

my $context = ZeroMQ::Context->new();

# Connect our subscriber socket
my $subscriber = $context->socket(ZMQ_SUB);
$subscriber->setsockopt(ZMQ_IDENTITY, 'Hello');
$subscriber->setsockopt(ZMQ_SUBSCRIBE, '');
$subscriber->connect('tcp://localhost:5565');

# Synchronize with publisher
my $sync = $context->socket(ZMQ_PUSH);
$sync->connect('tcp://localhost:5564');
$sync->send('');

# Get updates, expect random Ctrl-C death
while (1) {
    my $string = $subscriber->recv()->data;
    say $string;
    last if $string eq 'END';
}
